### PR TITLE
Type method refactor

### PR DESCRIPTION
**Pull request by ericperez**
I cleaned up the 'type' method a bit. The return path is a lot cleaner and the logic is a little easier to follow.

### CODE REVIEW

**Comment by zdfs** [comment]
This the tests pass, this looks good to me, but I'll let @LeaVerou merge this one.

File: bliss.shy.js [hunk 1]
```diff
@@ -54,16 +54,15 @@ extend($, {
...
-		if (obj === undefined) { return 'undefined'; }
-
-		var ret = (Object.prototype.toString.call(obj).match(/^\[object\s+(.*?)\]$/)[1] || "").toLowerCase();
-
-		if(ret == 'number' && isNaN(obj)) {
-			return 'nan';
+		var ret;
+		if(obj === null || obj === undefined) {
+			ret = obj + '';
+		} else {
```
Comment (LeaVerou) [inline bliss.shy.js]: Please follow the code style of the rest of the code. `else {` should be on a separate line.

@dperrymorrow is there a way to enforce this via jshint?

**Comment by LeaVerou** [comment]
Thanks for the contribution @ericperez!

Not sure why this is easier to follow though. If anything, it looks harder to follow to me:

1. `obj + ''` is an indirection: The reader needs to briefly look at what `obj` resolves to, and calculate what happens if you concat with a string to figure out what is happening. Sure, this won't take them more than a few seconds, but it's longer than what the current version would take.
2. Not returning as early as possible, but using a ret variable and returning at the end forces the reader to read the entire function to figure out what's going on, instead of stopping at the `return` statement one is interested in.

While I can find two reasons this is harder to follow, I can't see any that makes this easier to follow, except perhaps saving a few characters. Curious what your reasons were?

File: bliss.shy.js (hunk 1 above)
Comment (dperrymorrow) [inline bliss.shy.js]: not that I could find. there is a rule for ensuring whitespace around ```} else {``` but other than that i don't see anything.

**Comment by guoguo12** [comment]
Regardless of the consensus here, don't forget to run `gulp` to build the minified files!

**Comment by dperrymorrow** [comment]
actually thats a good question @guoguo12, do we need to create the dist files on each commit, or since we are now cutting releases, would that just be done at that point?

**Comment by guoguo12** [comment]
I didn't think of that. We should put it up for discussion!

**Comment by LeaVerou** [comment]
Since there's no response by the OP, I’m closing this.
@ericperez I hope a rejected PR does not discourage you from future contributions! While the specific PR was not merged, the willingness to contribute is always appreciated :)

**Pull request closed without merging**